src/consumer/consumer.js

```diff
@@ -0,0 +1,60 @@
+import Frame from './frame';
+
+/**
+ * Add the given event handler to an EventEmitter instance
+ * @param {EventEmitter} emitter - an instance of EventEmitter
+ * @param {string} event - event name
+ * @param {function} handler - event handler function
+ */
+function addEventHandler(emitter, event, handler) {
+  if (typeof handler === 'function') {
+    emitter.on(event, handler);
+  }
+}
+
+class Consumer {
+  /**
+   * Initialize a consumer.
+   * @param  {Array}  globalHandlers - an object containing event handlers that apply to all frames.
+   * @example
+   * // Each key/value pair in globalHandlers should be a pair of event name and event handler.
+   * const handlers = {'eventA': function() {}, 'eventB': function() {}};
+   * Consumer.init(handlers);
+   */
+  init(globalHandlers = {}) {
+    this.globalHandlers = globalHandlers;
+  }
+
+  /**
+   * Mount the given source as an application into the given container.
+   * @param {object} container - The DOM element to append the mounted frame to.
+   * @param {string} source - The source URL to load the app from.
+   * @param {string} options - An optional parameter that contains optional configs
+   * @return {Frame} Returns the application that was mounted.
+   */
+  mount(container, source, options = {}) {
+    const frame = new Frame();
+    frame.init(container, source, options);
+
+    // Apply global handlers to the frame
+    Object.keys(this.globalHandlers).forEach(event => {
+      const handlers = this.globalHandlers[event];
```

Comment:
I think we can guarantee this is an array by doing
```
const handlers = [].concat(this.globalHandlers[event])
```
This allow us to remove the conditional below and move the `addEventHandler` function inline into this code.

Reply:
We can also do something like:

```js
Object.entries(this.globalHandlers).forEach(([event, handler]) => {
  const handlers = [].concat(handler);
  ...
}
```

Reply:
Updated.